packages/typography/src/Link.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useMemo } from 'react';
+import ArrowRightIcon from '@leafygreen-ui/icon/dist/ArrowRight';
+import OpenNewTabIcon from '@leafygreen-ui/icon/dist/OpenNewTab';
+import Box, { ExtendableBox } from '@leafygreen-ui/box';
+import { useBaseFontSize } from '@leafygreen-ui/leafygreen-provider';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { uiColors } from '@leafygreen-ui/palette';
+import { HTMLElementProps, createDataProp } from '@leafygreen-ui/lib';
+import { typeScale1, typeScale2 } from './styles';
+
+const anchorDataProp = createDataProp('anchor-container');
+
+const linkStyles = css`
+  text-decoration: none;
+  color: ${uiColors.blue.base};
+  cursor: pointer;
+
+  &:focus {
+    outline: none;
+  }
+`;
+
+const underline = css`
+  background-repeat: repeat-x;
+  background-position: 0 1em;
+  background-size: 3px 3px;
+
+  ${anchorDataProp.selector}:hover & {
+    background-image: linear-gradient(
```

Comment:
I'm fine if we wait on this, but it would be great to transition the gradient in smoothly. That's not supported directly by CSS unfortunately so it'll probably take some JS to make the transition work in this case.

If you don't want to do this for v1, would you mind making a ticket?

Reply:
https://jira.mongodb.org/browse/PD-752